### PR TITLE
Pistol/revolver caliber guns balancing

### PR DESCRIPTION
**Pull request by HissyNays**
Adjusted .22, 9mm, 10mm, 14mm, .45, .357, .44, 45-70 and coresponding weapons to them to bring back relative competetiveness of mostly overlooked weapons. If my effort here changes anything, i'll also tweak rifle bullet weapons.

## Why It's Good For The Game
Brings back life to weapons that were otherwise always discarded thus creating variety of viable choices and loadouts to try, restores uniques that were otherwise as good as their common variants.
Returns small arms and lever action weapons to viability.

## Changelog

Increased damage for most pistol/revolver bullets with semi-auto in mind, decreased full auto weapons damage to balance out fire rate. The changes listed below will not be full as it takes a lot of time to list everything, this is mostly as example of what to expect.

=======AUTOMATIC=======

Decreased base slowdown for smgs: 0.4 to 0.25

10mm smg
-damage increased from 18 to 22

14mm smg
-damage decreased 32 to 27
-added armor penetration 0.15

Greasegun
-damage buffed from  16 to 19

Mini Uzi
-damage buffed from 16 to 17
-full auto fire delay from 2 to 1.80

Carl Gustaf (10mm)
-damage buffed from 18 to 24

Carl Gustaf (.45)
-damage buffed from  24 to 26

Thompson SMG
-damage buffed from 24 to 25

Storm Drum
-increased fire delay and spread slightly
-damage nerfed from 24 to 23

FN P90c
-full auto fire delay decreased slightly
-damage buffed from 18 to 22

worn FN P90c
-damage buffed from 18 to 20

MP-5 SD
-damage buffed from 16 to 21

PPSH-41
-damage buffed from 16 to 19

M1 Carbine (and it's subtypes)
-damage buffed from 18 to 22
-added extra bullet speed 100
-added extra armor penetration 0.05

M1/N Carbine
-damage buffed from 18 to 24

De Lisle Carbine
-damage buffed from 16 to 21
-re-added armor penetration 0.05

Commando carbine
-damage buffed from 24 to 25
-extra penetration re-added 0.2

Combat rifle
-damage buffed from 24 to 25

=======PISTOLS/REVOLVERS=======

Removed slowdown from .45 pistols, increased damage

.22 pistol
-buffed damage from 15 to 19

10mm pistol
-buffed damage from 21 to 22

Maria
-buffed damage from 24 to 25
-buffed armor penetration from 0.05 to 0.2

.357
-buffed damage from 28 to 35

.44 revolver
-buffed damage from 32 to 38
-added 0.1 ap

Browning hi-power
-buffed damage from 16 to 20

Beretta M93R
-Buffed damage from 16 to 21

M1911
-buffed damage from 24 to 30

Mk23
-buffed damage from 24 to 30
-Fire delay from 2 to 1

Desert Eagle 
-decreased damage from  44 to 39
-increased armor penetration from 0.12 to 0.15

Automag 
-increased damage from  37 to 40
-added 0.1 armor penetration

14mm pistol (and it's subtypes)
-decreased damage from 43 to 36
-added armor penetration 0.15
-added slowdown 0.05

Little devil (14mm)
-added armor penetration 0.3
-removed slowdown

hunting revolver
-damage buffed to 43
-added armor penetration 0.15
-added slowdown 0.05

=======RIFLES=======

Trail carbine 
-damage buffed from  36 to 39
-added armor penetration 0.1

Cowboy repeater
-buffed damage from 28 to 36

Brush gun
-damaged buffed from to 42 to 43
-armor penetration buffed from 0.12 to 0.15

Medicine stick (brush gun unique)
-extra speed from 150 to 100
-armor penetration from 0.4 to 0.25

(Some of changes listed in description are outdated)

### CODE REVIEW

**Comment by ProbablyCarl** [comment]
What's with the removal of the whitelist file?
Otherwise, this looks fairly good! I'll be running this by Dragon as well.

**Comment by Spc-Dragonfruits** [comment]
Maintainer requested review,

NO slowdown changes below 0.3 in SMGs, bar weak variants (9/10mm, 10mm Gustaf)
--While SMGs are generally regarded as fast moving spray and pray, having people run around like it's Quake with full auto code as is, is terrible

NO armor penetration higher than 12% in pistols
--Armor penetration is a primary trait of long-barreled guns like rifles, and large purpose built rounds like slugs.
ADDITIONALLY: The higher the extra penetration value, the more you should increase the "recoil" and "spread" variables. Desert Eagle for reference.

The Mk. 23's advantage is the suppressor and extra rounds, the damage buff makes it absurd
--It dominates through ammo capacity and modifiability as it stands

Lil Devil should have its slowdown reduced, no extra traits
--It is already absurdly strong, being a 12.7mm gun with a weight class of 1

NO damage buffs to SMGs, this heavily intrudes on the gameplay niche of automatic rifles
--If you want hard hitters, you have to expect to chew up more expensive ammo and be prepared to move considerably slower

The medicine stick should keep its speed. 30% penetration is apt. As it stands, there is only a single way to obtain it without admin intervention - and it's in a loadout set pit up against an AMR, long colts, and citykiller.

**Comment by HissyNays** [comment]
> What's with the removal of the whitelist file? Otherwise, this looks fairly good for a first time PR! I'll be running this by Dragon as well.

Was trying to test on my own server but didn't figure it out, whitelist file accidentally got into PR :x

**Comment by HissyNays** [comment]
> NO slowdown changes below 0.3 in SMGs, bar weak variants (9/10mm, 10mm Gustaf) --While SMGs are generally regarded as fast moving spray and pray, having people run around like it's Quake with full auto code as is, is terrible

-Agreed, 14mm smgs are more like baby bolters, it's more of overlook that they're left with 0.25 slowdown, otherwise 0.25 slowdown is still considerable amount, just not as drastic as for rifles in comparison

> NO armor penetration higher than 12% in pistols --Armor penetration is a primary trait of long-barreled guns like rifles, and large purpose built rounds like slugs. ADDITIONALLY: The higher the extra penetration value, the more you should increase the "recoil" and "spread" variables. Desert Eagle for reference.

- 45-70 hunting revolver has long rifle like ammo and biggest slowdown of all revolvers/pistols, also made a mistake on comment, slowdown on it is not 0.05, it's 0.12, I figured it's fair for it to be like that given it's pretty much pocket sniper rifle.
The 14mm is a large rare futuristic pistol, it could at the very least deserve a bit more AP and deagle's defining feature amidst other 44 caliber weapons was always that bit more AP. As for Maria, it's a rare unique and arguably worst one, I doubt it'll match up for any rifle or even common full auto shotgun with it's fairly big AP, it might aswell have it. It always did in the past and was never a problem.

> The Mk. 23's advantage is the suppressor and extra rounds, the damage buff makes it absurd --It dominates through ammo capacity and modifiability as it stands

-The current damage on Mk.23 is the damage it had before the nerf, even then it was always left in the open, not picked by anyone. Suppressor's advantage is very much negligible, it's mostly cosmetic as people you shoot at see the projectile flying at them and thus who fired at them or from which direction. Additionally blood splat from character when hit shows where you're hit from. The other remaining attachment you can put on it is recoil compensator which honestly changes absolutely nothing in practice for any semi-auto weapon. The only real attachment that might make difference is a flashlight by saving item slot.

The real advantage of Mk23 is that it's basicly m1911 with double capacity, it's a fairly rare weapon so I'd say it deserves it's damage, otherwise there will be no point in using it again.

> Lil Devil should have its slowdown reduced, no extra traits --It is already absurdly strong, being a 12.7mm gun with a weight class of 1

-Lildevil is the only 14mm pistol that has no slowdown, the only extra trait it has is extra 0.1 armor penetration, removing it might be too much as it's incredibly rare and was always best handgun in game, how about adjusting it to 0.05 extra armor penetration?

> NO damage buffs to SMGs, this heavily intrudes on the gameplay niche of automatic rifles --If you want hard hitters, you have to expect to chew up more expensive ammo and be prepared to move considerably slower

-This is less of a buff and more of a return to pre-unintentended nerf of smgs, SMG's as of right are kind of terrible and the damage buff doesn't really make them compete with assault rifles by a long shot. Only 14mm smg is comparable to assault rifles but it's incredibly rare. The other SMG comparable to assault rifles is thompson, but even then it is practicly already is assault rifle, just with a lack of extra bullet speed, It even has slowdown of a rifle. Why not give this a try and adjust accordingly?
> The medicine stick should keep its speed. 30% penetration is apt. As it stands, there is only a single way to obtain it without admin intervention - and it's in a loadout set pit up against an AMR, long colts, and citykiller.

-Brush gun has always been a desired weapon by everyone and Medicine stick is it's upgrade and still remains among absolute best weapons such as AMR even with nerf, but if it has anymore AP than what I suggest, it becomes better than any weapon listed in loadout sets.
As for bullet speed change it's more or less not to throw off aim from being used to using brush gun. It also makes very little sense that a same gun fires same bullets but bullet flies faster. The extra speed on it is very small, it shouldn't even be felt at all that it was nerfed, I'd suggest it stays.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 1]
```diff
@@ -238,9 +237,11 @@
 	w_class = WEIGHT_CLASS_BULKY
 	mag_type = /obj/item/ammo_box/magazine/smg14
 	is_automatic = TRUE
+	slowdown 0.35
```
Comment (Leon9621) [inline code/modules/projectiles/guns/ballistic/automatic.dm:240]: You're missing an = sign here, should be slowdown = 0.35

**Comment by Leon9621** [comment]
> -The current damage on Mk.23 is the damage it had before the nerf, even then it was always left in the open, not picked by anyone. Suppressor's advantage is very much negligible, it's mostly cosmetic as people you shoot at see the projectile flying at them and thus who fired at them or from which direction. Additionally blood splat from character when hit shows where you're hit from. The other remaining attachment you can put on it is recoil compensator which honestly changes absolutely nothing in practice for any semi-auto weapon. The only real attachment that might make difference is a flashlight by saving item slot.

> The real advantage of Mk23 is that it's basicly m1911 with double capacity, it's a fairly rare weapon so I'd say it deserves it's damage, otherwise there will be no point in using it again.

As someone who'd grab the Mk.23 9 / 10 when I saw it, the reason I did so was because I'd get more bullets per magazine, the convenience of freeing up a pocket from my flashlight besides. It already _has_ a niche in ammo capacity (and being able to carry more bullets in the same # of magazines as an M1911 for storage) - more damage isn't necessary imo.

 > This is less of a buff and more of a return to pre-unintentended nerf of smgs, SMG's as of right are kind of terrible and the damage buff doesn't really make them compete with assault rifles by a long shot. Only 14mm smg is comparable to assault rifles but it's incredibly rare. The other SMG comparable to assault rifles is thompson, but even then it is practicly already is assault rifle, just with a lack of extra bullet speed, It even has slowdown of a rifle. Why not give this a try and adjust accordingly?

Barring the American 180, 14mm, thompson / storm drum, and the PPSH, every smg fits in a backpack, and the first 3 of those have decent reasons why. The 180 has nutty ammo capacity, and the 14mm and thompson do pretty solid damage, with the thompson getting glorious drum magazines. I'd get into more but I gotta leave for work soon; these are my thoughts so far, as Just A Dude Who Plays The Game And Likes The Guns.

**Comment by HissyNays** [comment]
Yeah I gave Mk23 a thought again and it does seem like a really convenient weapon to carry even with less damage. For SMG's i'll try to make them head in the way of low damage,  high fire rate, low accuracy.

**Comment by Leon9621** [comment]
> Yeah I gave Mk23 a thought again and it does seem like a really convenient weapon to carry even with less damage. For SMG's i'll try to make them head in the way of low damage, high fire rate, low accuracy.

That's a good niche IMO! They're more portable than rifles and capable of more damage than pistols, exactly what they ought to be.

**Comment by Spc-Dragonfruits** [comment]
There's a buttload of undocumented changes in the PR body, please update it. Also, get rid of anything that's extra damage -1, 1, and 2. It creates useless clutter for no gameplay difference.

**Comment by HissyNays** [comment]
> There's a buttload of undocumented changes in the PR body, please update it. Also, get rid of anything that's extra damage -1, 1, and 2. It creates useless clutter for no gameplay difference.

People will always pick weapons with 1 or 2 more damage and with smg's 1 or 2 extra damage is big difference

**Comment by Leon9621** [comment]
> > There's a buttload of undocumented changes in the PR body, please update it. Also, get rid of anything that's extra damage -1, 1, and 2. It creates useless clutter for no gameplay difference.
> 
> People will always pick weapons with 1 or 2 more damage and with smg's 1 or 2 extra damage is big difference

I'm gonna keep it real with you, 1-2 more damage doesn't mean shit to me unless it hits some threshold for killing simplemobs in one less bullet.

This **ALSO** assumes that everyone who plays the game has read the code, knows + remembers the exact combination of damage values from bullets and guns, and added it all together in their head to know what's "worth picking" and what isn't. This is frankly ridiculous.

**Comment by HissyNays** [comment]
> > > There's a buttload of undocumented changes in the PR body, please update it. Also, get rid of anything that's extra damage -1, 1, and 2. It creates useless clutter for no gameplay difference.
> > 
> > 
> > People will always pick weapons with 1 or 2 more damage and with smg's 1 or 2 extra damage is big difference
> 
> I'm gonna keep it real with you, 1-2 more damage doesn't mean shit to me unless it hits some threshold for killing simplemobs in one less bullet.
> 
> This **ALSO** assumes that everyone who plays the game has read the code, knows + remembers the exact combination of damage values from bullets and guns, and added it all together in their head to know what's "worth picking" and what isn't. This is frankly ridiculous.

People who don't code dive can use common sense and will most likely be correct in assuming which weapon does more damage

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 2]
```diff
@@ -384,10 +388,11 @@
 	fire_delay = 3
 	burst_shot_delay = 2.2
 	is_automatic = TRUE
+	extra_damage = -5
 	automatic = 1
-	slowdown = 0.3
-	autofire_shot_delay = 2
-	spread = 16
+	autofire_shot_delay = 1.70
+	recoil - 1
```
Comment (Leon9621) [inline code/modules/projectiles/guns/ballistic/automatic.dm:394]: ...You forgot an = here as well. Unless this is a new change, either way- "recoil = -1"

**Comment by ProbablyCarl** [comment]
Stale.
I'll be closing this in a bit if there's no further responses.

File: code/modules/projectiles/guns/ballistic/automatic.dm (hunk 2 above)
Comment (HissyNays) [reply code/modules/projectiles/guns/ballistic/automatic.dm]: Thanks for the heads up

**Comment by ProbablyCarl** [comment]
Going to close this for the moment. Large changes locally planned. They override this to a significant degree and it'd just cause further issues to merge it now.

**Pull request closed without merging**